sdk/storage/Azure.Storage.Common/tests/StorageBearerTokenChallengeAuthorizationPolicyTests.cs

```diff
@@ -100,5 +102,46 @@ public async Task SendsUnUnAuthroizedRequestWhenEnableTenantDiscoveryIsTrue([Val
             // If enableTenantDiscovery is true, all but the first request should be authorized.
             Assert.True(transport.Requests.Skip(enableTenantDiscovery ? 1 : 0).All(r => r.Headers.TryGetValue("Authorization", out _)));
         }
+
+        [Test]
+        public async Task UsesScopeFromBearerChallange()
+        {
+            // Arrange
+            bool firstRequest = true;
+            string unauthorizedScope = "https://disk.compute.azure.com/.default";
+            string authorizedScope = "https://storage.azure.com";
```

Comment:
For clarity I would recommend naming these something like:
```suggestion
            string initialMismatchedScope = "https://disk.compute.azure.com/.default";
            string serviceChallengeResponseScope = "https://storage.azure.com";
```